src/core/server/http/cookie_session_storage.ts

```diff
@@ -98,15 +104,35 @@ export async function createCookieSessionStorageFactory<T>(
   cookieOptions: SessionStorageCookieOptions<T>,
   basePath?: string
 ): Promise<SessionStorageFactory<T>> {
+  function clearInvalidCookie(
+    req: Request | undefined,
+    path: string = basePath || '/',
+    isSecure: boolean = cookieOptions.isSecure
+  ) {
+    // if the cookie did not include the 'path' or 'isSecure' attributes in the session value, it is a legacy cookie
+    // we will assume that the cookie was created with the current configuration
+    log.debug(`Clearing invalid session cookie`);
+    // need to use Hapi toolkit to clear cookie with defined options
+    if (req) {
+      (req.cookieAuth as any).h.unstate(cookieOptions.name, { path, isSecure });
```

Comment:
Needed to add code to clear invalid cookies by specifying the `path` and `isSecure` attributes (using the Hapi toolkit).

Note, because of this, we no longer need Hapi to automatically clear invalid cookies (see below).